File: DynamicAppConfigAndroid/DynamicAppConfigExample/src/main/java/com/crescentflare/dynamicappconfigexample/appconfig/ExampleAppConfigRunType.java
package com.crescentflare.dynamicappconfigexample.appconfig;

/**
 * App config: application configuration run setting
 * An enum used by the application build configuration
 */
public enum ExampleAppConfigRunType
{
    RunNormally("runNormally"),
    RunQuickly("runQuickly"),
    RunStrictly("runStrictly");

    private String text;

    ExampleAppConfigRunType(String text)
    {
        this.text = text;
    }

    public static ExampleAppConfigRunType fromString(String text)
    {
        if (text != null)
            for (ExampleAppConfigRunType e : ExampleAppConfigRunType.values())
                if (text.equalsIgnoreCase(e.text))
                    return e;
        return RunNormally;
    }

    public String toString()
    {
        return text;
    }
}

File: DynamicAppConfigIOS/Example/DynamicAppConfig/AppConfig/ExampleAppConfigRunType.swift
//
//  ExampleAppConfigRunType.swift
//  DynamicAppConfig Example
//
//  App config: application configuration run setting
//  An enum used by the application build configuration
//

import UIKit
import DynamicAppConfig

// Enum definition using string raw values for storage
enum ExampleAppConfigRunType: String, CaseIterable {
    
    case runNormally = "runNormally"
    case runQuickly = "runQuickly"
    case runStrictly = "runStrictly"

}
